doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        winrt::Windows::UI::Xaml::Controls::Button cancelButton,
+        winrt::Windows::UI::Xaml::Controls::ProgressBar progressBar,
+        winrt::Windows::UI::Xaml::Controls::TextBlock statusText)
+    {
+        installButton.IsEnabled(false);
+        cancelButton.IsEnabled(true);
+
+        co_await winrt::resume_background();
+
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog catalog{ appInstaller.GetAppCatalog(PredefinedAppCatalog::OpenWindowsCatalog) };
+        catalog.OpenAsync().get();
+        //CatalogPackage package{ FindPackageInCatalog(catalog, m_installAppId).get() };
+        FindPackagesResult findPackagesResult{ TryFindPackageInCatalog(catalog, m_installAppId).get() };
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        if (matches.Size() > 0)
+        {
+            m_installPackageOperation = InstallPackage(matches.GetAt(0).CatalogPackage());
+            UpdateUIForInstall(m_installPackageOperation, installButton, cancelButton, progressBar, statusText);
+        }
+    }
+
+```
+
+## 3.3.1 Cancel
+
+The async operation must be stored, or the install code must wait on an event that can be triggered.
+
+```c++ (C++ish pseudocode)
+    void MainPage::CancelButtonClickHandler(IInspectable const&, RoutedEventArgs const&)
+    {
+        if (m_installPackageOperation)
+        {
+            m_installPackageOperation.Cancel();
+        }
+    }
+```
+
+## 3.3.2. Cancel
+
+Cancel the async operation
+
+```c++ (C++ish pseudocode)
+    IAsyncAction CancelInstall(
+        std::wstring installAppId)
+    {
+        co_await winrt::resume_background();
+        // Creation of the AppInstaller has to use CoCreateInstance rather than normal winrt initialization since it's created 
+        // by an out of proc com server.
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog windowsCatalog{ appInstaller.GetAppCatalog(PredefinedAppCatalog::OpenWindowsCatalog) };
+        windowsCatalog.OpenAsync().get();
+        GetCompositeAppCatalogOptions getCompositeAppCatalogOptions = CreateGetCompositeAppCatalogOptions();
+        getCompositeAppCatalogOptions.Catalogs().Append(windowsCatalog);
+        getCompositeAppCatalogOptions.LocalAppCatalog(LocalAppCatalog::InstallingPackages);
+        // Specify that the search behavior is to only query for local packages.
+        // Since the local catalog that is open is InstallingPackages, this will only find a result if installAppId is 
+        // currently installing.
+        getCompositeAppCatalogOptions.CompositeSearchBehavior(CompositeSearchBehavior::InstallingPackages);
+        AppCatalog compositeCatalog{ appInstaller.GetCompositeAppCatalog(getCompositeAppCatalogOptions) };
+        co_await compositeCatalog.OpenAsync();
+
+        FindPackagesOptions findPackagesOptions = CreateFindPackagesOptions();
+        PackageMatchFilter filter;
+        filter.IsAdditive(true);

```

Comment:
Is IsAdditive to be used for constructing AND/OR queries? It's not clear to me how to construct say:

(Name StartsWith "Office" OR Name StartsWith "Photo") AND AppCatalogDefined Exact "true"
vs
Name StartsWith "Office" OR (Name StartsWith "Photo" AND AppCatalogDefined Exact "true")

Reply:
Yes, IsAdditive true indicates an OR filter, false indicates an AND filter. Perhaps a FilterType enum with Or and And is clearer? Or just separate vectors? There are some gotchas here that are described in the idl section down at line ~700. Specifically, due to the underlying code implementation that this api is hooking up to the order of the Filters vector does not matter and only one filter with AppCatalogDefined match is allowed.